### DISCUSSION TITLE
Error when running with pg_connect_pool

### DISCUSSION THREAD

**Discussion by eqbalz** [TPC-Council/HammerDB#689] [Q&A] [answered]
I am trying to do a TPC-C run with Postgres cluster and directing read queries to standby (HammerDB-4.4). But when running with `pg_connect_pool`, the schema build succeeds but I get an error when running the load.

```
Vuser 2:prep_statement 3: prepare neword (INTEGER, INTEGER, INTEGER, INTEGER, INTEGER) as select neword($1,$2,$3,$4,$5,0)
Error in Virtual User 2: ERROR:  function neword(integer, integer, integer, integer, integer, integer) does not exist
LINE 1: ...ER, INTEGER, INTEGER, INTEGER, INTEGER) as select neword($1,...
                                                             ^
HINT:  No function matches the given name and argument types. You might need to add explicit type casts.

Vuser 2:FINISHED FAILED
Vuser 3:RUNNING
Vuser 3:prep_statement 3: prepare neword (INTEGER, INTEGER, INTEGER, INTEGER, INTEGER) as select neword($1,$2,$3,$4,$5,0)
Error in Virtual User 3: ERROR:  function neword(integer, integer, integer, integer, integer, integer) does not exist
LINE 1: ...ER, INTEGER, INTEGER, INTEGER, INTEGER) as select neword($1,...
```

I have tried to run both with `pg_storedprocs` set to `true` as well as `false` and get the same error.
Any help is appreciated.

Thanks.

**Comment by sm-shaw** [comment, marked as answer]
Yes, you need to set pg_storedprocs to false for both the schema build and the test. As described here https://www.hammerdb.com/docs/ch04s09.html prepared statements are not supported for the connect pool option as it uses prepared statements and these are not supported by PostgreSQL with stored procedures and therefore you have to both build and run with functions. 
Also there is this issue https://github.com/TPC-Council/HammerDB/issues/681 that has been fixed with the PostgreSQL connect pool implementation - so if you see this error add an additional close brace "}" as a workaround.
